### PR TITLE
Add links and examples for std::process::Stdio

### PR DESCRIPTION
**Pull request by mistodon**
As per #29370

### CODE REVIEW

File: src/libstd/process.rs [hunk 1]
```diff
@@ -742,21 +742,128 @@ impl fmt::Debug for Output {
     }
 }
 
-/// Describes what to do with a standard I/O stream for a child process.
+/// Describes what to do with a standard I/O stream for a child process when
+/// passed to the [`stdin`], [`stdout`], and [`stderr`] methods of [`Command`].
+///
+/// [`stdin`]: ./struct.Command.html#method.stdin
```
Comment (dtolnay) [inline src/libstd/process.rs]: I don't know much about rustdoc but these are the only instances of `grep -r '\./struct'` in the whole repo. Could you check whether other docs use a different style for links?

**Comment by dtolnay** [comment]
```
[00:04:33] tidy error: /checkout/src/libstd/process.rs:786: trailing whitespace
[00:04:33] tidy error: /checkout/src/libstd/process.rs:791: trailing whitespace
[00:04:33] tidy error: /checkout/src/libstd/process.rs:821: trailing whitespace
[00:04:33] tidy error: /checkout/src/libstd/process.rs:827: trailing whitespace
[00:04:33] tidy error: /checkout/src/libstd/process.rs:857: trailing whitespace
[00:04:33] tidy error: /checkout/src/libstd/process.rs:863: trailing whitespace
[00:04:34] some tidy checks failed
```

File: src/libstd/process.rs (hunk 1 above)
Comment (Pirh) [reply src/libstd/process.rs]: Sure, I'll take a look and I'll fix the trailing whitespace.

**Comment by Pirh** [comment]
Judging by other examples in the docs, the `./` prefix isn't used, so I've removed it. Also fixed the tidy errors.

**Comment by dtolnay** [comment]
Thanks! Great work on these examples.

@bors r+

**Pull request merged**